client/src/components/settings/about.tsx

```diff
@@ -205,14 +206,15 @@ class AboutSettings extends Component<AboutProps, AboutState> {
       toggleSoundMode,
       toggleKeyboardShortcuts
     } = this.props;
+    const ariaLabel = t('settings.headings.personal-info');
     return (
       <>
         <UsernameSettings username={username} />
         <Spacer size='medium' />
         <SectionHeader>{t('settings.headings.personal-info')}</SectionHeader>
         <FullWidthRow>
           <form id='camper-identity' onSubmit={this.handleSubmit}>
-            <div role='group' aria-label={t('settings.headings.personal-info')}>
+            <div role='group' aria-label={ariaLabel}>
```

Comment:
That issue with new types, the return doesn't include `undefined`, the type of `aria-label` doesn't align with the code below, I can open issue because this is a valid issue 

```ts
DefaultTFuncReturn = string | (TypeOptions['returnNull'] extends true ? null : never)
```

Reply:
Btw why is aria-label type `string | undefined`, shouldn't it be always a `string`? 🤔

Reply:
I won't lie both solutions doesn't feel nice, because these feels like Types adding more context instead of guiding use to the correct context, I prefer the first one more because:

- it's type align with `TFunction` and `aria-label`, as one wants to produce string and other wants to receive string, so they are aligning.
- and to be honest, I don't lazy 🤣.

---

I will try to create sandbox and convice the `react-i18next` team to include `undefined` in the return type.

Reply:
So, I dug into this a little. The issue is, I think, that you're expected to merge your `CustomTypeOptions` into `i18next` per: https://www.i18next.com/overview/typescript.

If you do that, it knows all your keys and will not only give you correct return types (`string` not `string | null`), but using the wrong keys will raise type errors. It's neat.

...unfortunately this doesn't work because we use `preval` to make sure that we do not bundle up all the .json files whenever we build the client for a single language.  It's possible that the state of bundling has moved on since we did this and we can ditch `preval` and just import the JSON. I'm not sure, but it could be worth investigating.

Reply:
Let's come back to this after https://github.com/freeCodeCamp/freeCodeCamp/pull/49766 is merged, I will create an issue for this when the PR is merged. 👍

Reply:
Do you want to go with these PR's as they are then @Sboonny? or fix the type issue and remove these changes from PR's like this?